Gems/EMotionFX/Code/Source/Integration/Components/SimpleMotionComponent.cpp

```diff
@@ -307,8 +308,23 @@ namespace EMotionFX
         {
             if (m_motionInstance)
             {
-                float delta = time - m_motionInstance->GetLastCurrentTime();
-                m_motionInstance->SetCurrentTime(time, false);
+                //PlayTime setting precision.
+                //The play time of the motion in the animation editor is displayed as a rounded two decimal places.
+                //Therefore, you need to determine the precision of the input parameter.
+                const float playTimePrecision = 0.005f;
+                float revisedTime = time;
+                if (MCore::Compare<float>::CheckIfIsClose(time, m_motionInstance->GetDuration(), playTimePrecision) == true)
+                {
+                    revisedTime = m_motionInstance->GetDuration();
+                }
+                else if (revisedTime > (m_motionInstance->GetDuration() + MCore::Math::epsilon))
+                {
+                    AZ_Warning("EMotionFX", false, "Invalid play time: %f (max:%.2f), The set value exceeds the maximum value.", time, m_motionInstance->GetDuration());
+                    return;
+                }
+
+                float delta = revisedTime - m_motionInstance->GetLastCurrentTime();
+                m_motionInstance->SetCurrentTime(revisedTime, false);
```

Comment:
If you need higher precision on the playtime, would it help to just increase the number of decimals displayed in the UI? To let's say milliseconds precision?